### PR TITLE
clean normalization_inner

### PR DESCRIPTION
**Pull request by liqiangxl**
(1) Transformed private static functions in the InnerPersistentKernelScheduler class into utility functions within an anonymous namespace.
(2) Removed checks/calculations that are unrelated to inner persistent scheduler.

### CODE REVIEW

File: csrc/scheduler/normalization_inner.cpp [hunk 1]
```diff
@@ -5,78 +5,146 @@
...
+  FUSER_PERF_SCOPE("Schedule InnerPersistent Fusion");
   scheduleInnerPersistentKernel(fusion, reductionParams());
 }
 
+namespace {
+
+bool checkReductionPattern(
+    Fusion* fusion,
+    const std::vector<TensorView*>& reduction_tvs) {
+  // Use root domain map to check the reduction ops have the same axes
```
Comment (liqiangxl) [inline csrc/scheduler/normalization_inner.cpp:38]: (1) Transformed from a static function in the InnerPersistentKernelScheduler class into a utility function within an anonymous namespace. 
(2) Removed checks that were unrelated to inner persistence.

File: csrc/scheduler/normalization_inner.cpp [hunk 2]
```diff
@@ -5,78 +5,146 @@
...
+    }
+  }
+  return true;
+}
+
+std::pair<int64_t, int64_t> getPersistentBufferSize(
+    Fusion* fusion,
+    SchedulerRuntimeInfo& runtime_info,
+    HeuristicSummary* data_cache,
+    const std::vector<TensorView*>& reduction_tvs) {
```
Comment (liqiangxl) [inline csrc/scheduler/normalization_inner.cpp:63]: (1) Transformed from a static function in the InnerPersistentKernelScheduler class into a utility function within an anonymous namespace.
(2) Removed checks that were unrelated to inner persistence.

File: csrc/scheduler/normalization_inner.cpp [hunk 3]
```diff
@@ -86,54 +154,39 @@ bool InnerPersistentKernelScheduler::canScheduleCompileTime(Fusion* fusion) {
...
-    }
-  }
-  bool combined_inner_outer =
-      !inner_reduction_tvs.empty() && !outer_reduction_tvs.empty();
+  auto reduction_type =
+      reduction_scheduler_utils::getReductionType(reduction_tvs);
 
-  if (inner_reduction_tvs.empty() || !outer_reduction_tvs.empty()) {
+  if (reduction_type != reduction_scheduler_utils::ReductionType::Inner) {
     scheduler_debug_utils::canScheduleRejectReason(
```
Comment (liqiangxl) [inline csrc/scheduler/normalization_inner.cpp:165]: simplify reduction type check using utility function.

File: csrc/scheduler/normalization_inner.cpp [hunk 4]
```diff
@@ -251,70 +282,25 @@ bool InnerPersistentKernelScheduler::canScheduleRunTime(
...
-  }
-
-  const int64_t norm_per_sm =
-      ceilDiv(scheduler_utils::register_file_size, persistent_buffer_size);
-
-  // If outer reduction, don't go persistent if we can't fit half a warp in
-  // the iter domain of the persistent reduction.
-  if (!properties.fastest_dim_reduction &&
-      !(norm_per_sm >= warp_size / 2 ||
-        max_multi_reduction_factor >= warp_size)) {
```
Comment (liqiangxl) [inline csrc/scheduler/normalization_inner.cpp:315]: Remove checks not used by inner persistent.

File: csrc/scheduler/normalization_inner.cpp [hunk 5]
```diff
@@ -251,70 +282,25 @@ bool InnerPersistentKernelScheduler::canScheduleRunTime(
...
-  const int64_t max_multi_reduction_factor = scheduler_utils::safeDiv(
-      available_persistent_buffer_size, persistent_buffer_size);
 
   const int64_t required_sm_per_norm =
       ceilDiv(persistent_buffer_size, scheduler_utils::register_file_size);
 
   // If the persistence requires over half the device don't do grid
   // persistence as we can't overlap the grid comms.
   if (required_sm_per_norm >
-      scheduler_utils::safeDiv(device_multiprocessor_count, 3)) {
```
Comment (liqiangxl) [inline csrc/scheduler/normalization_inner.cpp:301]: change from `3` to `2` based on the comments.

File: csrc/scheduler/normalization_inner.cpp [hunk 6]
```diff
@@ -345,648 +331,8 @@ void InnerPersistentKernelScheduler::computeHeuristics(
...
-    available_persistent_buffer_size = std::max(
-        available_persistent_buffer_size, available_shared_memory_size);
-  }
-
-  return std::make_pair(
-      persistent_buffer_size, available_persistent_buffer_size);
-}
-
-bool InnerPersistentKernelScheduler::canScheduleRunTimeOuter(
-    Fusion* fusion,
```
Comment (liqiangxl) [inline csrc/scheduler/normalization_inner.cpp:491]: deleted.

File: csrc/scheduler/normalization_inner.cpp [hunk 7]
```diff
@@ -1592,382 +938,26 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
...
-            << "\n"
-            << "max_multi_reduction_factor: " << max_multi_reduction_factor
-            << "\n"
-            << "block(" << hp.bdimx.get() << ", " << hp.bdimy.get() << ", 1)"
-            << std::endl;
-    debug() << rparams->toString() << std::endl;
-  }
-
-  return rparams;
-}
```
Comment (liqiangxl) [inline csrc/scheduler/normalization_inner.cpp:1929]: heuristics for outer and innerOuter are removed.

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM. Just in case, please do the diff check. It's currently disabled in CI, so needs to be done manually.

**Comment by liqiangxl** [comment]
Previous HEAD position was 8facc545 Define `TensorMap` for CUDA 11 (#932)
Switched to branch 'llu/clean_inner_norm'
Your branch is up to date with 'origin/llu/clean_inner_norm'.


DIFF RESULT:

No difference found
Already on 'llu/clean_inner_norm'
Your branch is up to date with 'origin/llu/clean_inner_norm'.

**Comment by naoyam** [comment]
> Previous HEAD position was [8facc54](https://github.com/NVIDIA/Fuser/commit/8facc5452ab62bd0180d9519b4dab4939b03481f) Define `TensorMap` for CUDA 11 (#932) Switched to branch 'llu/clean_inner_norm' Your branch is up to date with 'origin/llu/clean_inner_norm'.
> 
> DIFF RESULT:
> 
> No difference found Already on 'llu/clean_inner_norm' Your branch is up to date with 'origin/llu/clean_inner_norm'.

Thanks. Just please make sure to run the benchmarks as I don't think the script runs them by default.

**Comment by liqiangxl** [comment]
> > Previous HEAD position was [8facc54](https://github.com/NVIDIA/Fuser/commit/8facc5452ab62bd0180d9519b4dab4939b03481f) Define `TensorMap` for CUDA 11 (#932) Switched to branch 'llu/clean_inner_norm' Your branch is up to date with 'origin/llu/clean_inner_norm'.
> > DIFF RESULT:
> > No difference found Already on 'llu/clean_inner_norm' Your branch is up to date with 'origin/llu/clean_inner_norm'.
> 
> Thanks. Just please make sure to run the benchmarks as I don't think the script runs them by default.

The CI runs the benchmarks. e.g. https://gitlab-master.nvidia.com/dl/pytorch/fuser-gh-mirror/-/jobs/69890090

**Comment by naoyam** [comment]
I think the diff check is not enabled at this moment.

**Comment by liqiangxl** [comment]
> I think the diff check is not enabled at this moment in the CI.

You are right. The diff check is not enabled in CI. But CI runs the benchmark `Job jit_nvfuser_bench_jitfuture_TNVF` and I ran the `tools/compare_codegen.sh` on my local node.

**Comment by naoyam** [comment]
> > I think the diff check is not enabled at this moment in the CI.
> 
> You are right. The diff check is not enabled in CI. But CI runs the benchmark `Job jit_nvfuser_bench_jitfuture_TNVF` and I ran the `tools/compare_codegen.sh` on my local node.

What I wanted to make sure was to run the benchmarks with the diff script as it doesn't by default.

**Comment by liqiangxl** [comment]
> > > I think the diff check is not enabled at this moment in the CI.
> > 
> > 
> > You are right. The diff check is not enabled in CI. But CI runs the benchmark `Job jit_nvfuser_bench_jitfuture_TNVF` and I ran the `tools/compare_codegen.sh` on my local node.
> 
> What I wanted to make sure was to run the benchmarks with the diff script as it doesn't by default.

got you! I did this check in https://github.com/NVIDIA/Fuser/pull/928 using `tools/compare_codegen.sh -- build/nvfuser_bench --benchmark_filter=NvFuserScheduler --benchmark_repetitions=1 --benchmark_min_time=0`. No diff found.

**Pull request merged**